### PR TITLE
fix(metrics): Utilize controller-runtime  metrics server

### PR DESCRIPTION
**Pull request by grzesuav**
Signed-off-by: grzesuav <grzesuav@gmail.com>

### CODE REVIEW

**Comment by fpetkovski** [comment]
Are we keeping the existing metrics with this change?

**Comment by grzesuav** [comment]
> Are we keeping the existing metrics with this change?

Even better, we are adding them. Currently very few metrics were exposed, as we use the same k8s client for controller-runtime and metacontroller, and controller-runtime registeres it right, we got all worqueue metric in place ! I will add some example metrics before and after the change

File: main.go [hunk 1]
```diff
@@ -120,26 +114,7 @@ func main() {
...
-		}
-		klog.InfoS("Stopped metrics server")
-	}()
-
-	mgrStopChan.Done()
-	if err = srv.Shutdown(context.Background()); err != nil {
-		klog.ErrorS(err, "Error shutting down...")
-	}
+	<-mgrStopChan.Done()
+	klog.InfoS("Stopped metrics server")
```
Comment (fpetkovski) [inline main.go:118]: Should be updated to something like Stopped controller manager or similar

**Comment by fpetkovski** [comment]
Ah that's great! Just one nit, otherwise lgtm

**Comment by grzesuav** [comment]
[new.txt](https://github.com/metacontroller/metacontroller/files/6703316/new.txt)
[old.txt](https://github.com/metacontroller/metacontroller/files/6703317/old.txt)

I ran `examples/configmappropagation` test and collected metrics afterards.

**Pull request merged**